### PR TITLE
Coverage for Cypress tests

### PR DESCRIPTION
**Pull request by jenny-s51**
PR for #3199

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ version: 2
 # I'd love to find docs on this syntax.
 references:
   js_deps_paths: &js_deps_paths
+  - /home/circleci/cache/Cypress
```
Comment (redallen) [inline .circleci/config.yml]: Don't need this now

File: .circleci/config.yml [hunk 2]
```diff
@@ -26,7 +27,7 @@ references:
   - packages/react-codemods/node_modules/
   - packages/react-icons/node_modules/
   build_cache_paths: &build_cache_paths
-  - .cache/
+  - /home/circleci/cache/Cypress
```
Comment (redallen) [inline .circleci/config.yml]: Don't need this now

File: .circleci/config.yml [hunk 3]
```diff
@@ -141,8 +142,11 @@ jobs:
         paths: *build_cache_paths
         key: *build_cache_key
     - persist_to_workspace:
-        root: ~/project
-        paths: *build_cache_paths
```
Comment (redallen) [inline .circleci/config.yml]: Leave this in

File: .circleci/config.yml [hunk 4]
```diff
@@ -141,8 +142,11 @@ jobs:
...
+    #for integration tests
+        root: ~/
```
Comment (redallen) [inline .circleci/config.yml]: Add new `- persist_to_workspace:`

File: yarn.lock [hunk 5]
```diff
@@ -1,4 +1,5 @@
 # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+
```
Comment (redallen) [inline yarn.lock]: You can revert this now

File: .circleci/config.yml [hunk 6]
```diff
@@ -182,6 +186,17 @@ jobs:
...
+        name: Run Cypress Integration Tests
+        command: yarn test:integration && yarn run codecov -f packages/patternfly-4/react-integration/results/merged-test-results.xml -F patternfly4integration
+        no_output_timeout: 180m
+    - run: 
+        name: Create HTML file of Cypress coverage
+        command: yarn build:integration:report
+    - persist_to_workspace:
+        root: ~/project
+        paths:
+        - packages/patternfly-4/react-integration/results
```
Comment (redallen) [inline .circleci/config.yml]: Please make https://github.com/patternfly/patternfly-react/blob/master/.circleci/copy-docs.sh copy this HTML file to the PF4 docs directory! Then it'll get uploaded along with a doc build.

Comment (redallen) [reply .circleci/config.yml]: Optionally, also store the folder using
```yml
- store_artifacts:
          path: /tmp/artifacts
```
so we can download it from CircleCI if we ever need to! Docs: https://circleci.com/docs/2.0/artifacts/

File: .circleci/config.yml [hunk 7]
```diff
@@ -26,7 +26,6 @@ references:
   - packages/react-codemods/node_modules/
   - packages/react-icons/node_modules/
   build_cache_paths: &build_cache_paths
-  - .cache/
```
Comment (redallen) [inline .circleci/config.yml]: You can put this back, it's related to Gatsby, not Cypress.

File: .circleci/config.yml [hunk 8]
```diff
@@ -182,6 +187,19 @@ jobs:
     - run:
```
Comment (redallen) [inline .circleci/config.yml]: After `- *attach_workspace` add:
```
- attach_workspace:
      at: ~/
```
and I think that will restore the workspace you cached!

**Comment by evwilkin** [comment]
Resolves #3199

**Pull request merged**